src/core/meshcop/timestamp.cpp

```diff
@@ -62,16 +58,30 @@ int Timestamp::Compare(const Timestamp *aFirst, const Timestamp *aSecond)
 
     // Both are non-null.
 
-    firstSeconds  = aFirst->GetSeconds();
-    secondSeconds = aSecond->GetSeconds();
+    rval = Compare(*aFirst, *aSecond);
+
+exit:
+    return rval;
+}
+
+int Timestamp::Compare(const Timestamp &aFirst, const Timestamp &aSecond)
+{
+    int      rval;
+    uint64_t firstSeconds;
+    uint64_t secondSeconds;
+    uint16_t firstTicks;
+    uint16_t secondTicks;
+
+    firstSeconds  = aFirst.GetSeconds();
+    secondSeconds = aSecond.GetSeconds();
 
     if (firstSeconds != secondSeconds)
     {
         ExitNow(rval = (firstSeconds > secondSeconds) ? 1 : -1);
     }
 
-    firstTicks  = aFirst->GetTicks();
-    secondTicks = aSecond->GetTicks();
+    firstTicks  = aFirst.GetTicks();
```

Comment:
should we also compare the authoritative bit?

Reply:
Was trying to keep the existing logic unchange but simply expose the full Timestamp. The question may be, does the `authoritative` bit takes higher precedence that seconds and ticks?
@abtink @librasungirl thoughts?

Reply:
Previously active timestamp was used more like a priority value of a dataset. The larger, the higher precedence. In this perspective, I think we should take this bit into account. At least in case of the seconds and ticks are equal.

Reply:
Given the concept that `Active timestamp: a time value assigned to the Active Operational Dataset and used for comparing the priority between different Active Operational Datasets`

LGTM to only compare the time value portions.

Reply:
If authoritative timestamp is preferred, I'm inclined to not ignore the bit when comparing. Besides, take it into account can simplify the implementation as well. Just compare the whole 64-bit value.

Reply:
> Besides, take it into account can simplify the implementation as well. Just compare the whole 64-bit value.

We don't get the uint64_t any way (or do we want to change current `MeshCoP::Timestamp` implementation to use a uint64_t as underlying data?

Reply:
I agree it would be okay to take the authoritative bit into account, at least as the least significant bit. I updated the Timestamp class implementation to use a uint64_t underlying value. PTAL

Reply:
Thread Specification states:
```
Note that the U bit is the least significant bit when the entire timestamp is represented as a single 64-bit number.
```

Reply:
Sorry I reverted it when reverting the `uint64_t` changes, added it back in the last commit https://github.com/openthread/openthread/pull/7739/commits/96198fc70ee05769f523584848bc0eec51498fac